experimental/oak_functions_loader_linux_vsock/src/tests.rs

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::channel::Channel;
+use assert_matches::assert_matches;
+use oak_baremetal_communication_channel::{Read, Write};
+use std::io::Cursor;
+
+const TEST_BUFFER_SIZE: usize = 32;
+const TEST_MESSAGE: [u8; 8] = [0, 1, 2, 3, 4, 5, 6, 7];
+
+#[test]
+fn test_channel() {
+    let mut write_buffer = vec![0; TEST_BUFFER_SIZE];
+
+    let write_stream = Cursor::new(&mut write_buffer);
+    let mut write_channel = Channel::new(write_stream);
+
+    let write_result = write_channel.write(&TEST_MESSAGE);
+    assert_matches!(write_result, Ok(_));
+
+    // Copy data to a new stream and remove trailing zeroes.
+    let mut read_buffer: Vec<u8> = write_buffer.into_iter().rev().skip_while(|&x| x == 0).collect();
```

Comment:
This seems potentially flaky. If you are using a Vec anyway, can you not just start with a mutable vec for the buffer? I would assume that it would then have the correct size once writing has finished and you can reuse it for the read cursor.

Reply:
I don't think `Cursor` is going to increase the buffer size.

Reply:
Based on the comment in https://doc.rust-lang.org/std/io/struct.Cursor.html#method.new I think it will overwrite a mutable vector if it is not empty, but append if it is empty.